Automattic/wp-calypso
# Stats: Updating empty state for Number of Shares

Related to https://github.com/Automattic/red-team/issues/101

## Proposed Changes

* adding empty state, loader and info popover for Number of Shares
* adding TypeScript

## Why are these changes being made?


* it's part of the new empty states project

## Testing Instructions



* navigate to the live branch and verify that the `Number of shares` module works without the feature flag
* apply the flag `stats/empty-module-v2`
* check for a blog with and without data

![SCR-20240719-ngbm](https://github.com/user-attachments/assets/8e473264-b3ae-47ae-93e4-04fef2e4a59e)


## Pre-merge Checklist
